fix unnecessary item switching to velour equipment by buffMaintain

taltamir opened the pull request:
My previous implementation of velour item switching as part of buffmaintain resulted in unnecessary flip flopping on equipment even when we were not casting the buff. I only accounted for speculative but not for things like "we already have the buff".

## Tested:
ash calls. a bunch of standard runs in various classes.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have based by pull request against the [master branch](https://github.com/Loathing-Associates-Scripting-Society/autoscend/tree/master) or have a good reason not to.

taltamir commented:
```
> ash import autoscend; buffMaintain($effect[scariersauce])

Returned: false

> ash import autoscend; uneffect($effect[scariersauce])

Scariersauce has no specific item or skill for removal. Trying generics...
Using soft green whatever...
Scariersauce removed.
Returned: true

> ash import autoscend; buffMaintain($effect[scariersauce])

Putting on velour viscometer...
Equipment changed.
Casting Scarysauce 1 times...
You acquire an effect: Scariersauce (15)
Scarysauce was successfully cast.
Returned: true

> ash import autoscend; uneffect($effect[scariersauce])

Scariersauce has no specific item or skill for removal. Trying generics...
Using soft green whatever...
Scariersauce removed.
Returned: true

> ash import autoscend; buffMaintain($effect[scariersauce])

Casting Scarysauce 1 times...
You acquire an effect: Scariersauce (15)
Scarysauce was successfully cast.
Returned: true

> ash import autoscend; buffMaintain($effect[scariersauce])

Returned: false
```

Malibu-Stacey commented:
No offence but this is why I said you shouldn't import autoscend in your personal scripts. This change has no use in autoscend but now we need to maintain it.

taltamir commented:
> No offence but this is why I said you shouldn't import autoscend in your personal scripts. This change has no use in autoscend but now we need to maintain it.

While I found the error and fixed it while tinkering with my crimbo script. These are not for my personal project.

These are new buffs added to the game in 2021, we normally support new content when we can.
I only used 1 out of the 3 of them in my personal script. And I have added their use in various locations in autoscend.

#992 the issue description
#993 the PR where i added them
As you can see there the buffs are called via providers, community service, heavy rains, exploathing, level 3 quest, level 9 quest,  level 13 quest, and sauceress tower.

6 cold res and 6 sleaze res can save quite a lot of turns in autoscend. As well as resolve inability to progress in 0 IOTM accounts during standard. And will be used by autoscend from now on.

Actually I have been thinking about some more places to use it. If we auto pull the necessary items in certain conditions it will increase when we use them (currently only in casual and postronin). By writing a function to see how many adv they save and then auto pulling them if they save adv they would also be used in normal. (standard or older paths).

taltamir commented:
Question. Currently I am not bothering to switch the outfit back after using the buff because I am not setting the outfit as desireable and thus assuming it will get automatically switched away by auto_pre_adventure when it runs. Should I switch the outfit back for cleaner implementation?

quarklikeadork commented:
> Question. Currently I am not bothering to switch the outfit back after using the buff because I am not setting the outfit as desireable and thus assuming it will get automatically switched away by auto_pre_adventure when it runs. Should I switch the outfit back for cleaner implementation?

I think yes if buffs can be called after other equipment is selected for something that doesn't go through pre adventure, maybe tower contest registration with spooky damage?

taltamir commented:
tested with automatic restoring of equipment to original state
```
> pull velour viscometer

Pulling items from storage...
4 pulls remaining, 0 budgeted for automatic use.

> ash import autoscend; buffMaintain($effect[scariersauce])

Putting on velour viscometer...
Equipment changed.
Casting Scarysauce 1 times...
You acquire an effect: Scariersauce (5)
Scarysauce was successfully cast.
Putting on numberwang...
Equipment changed.
Returned: true
```
